From 1c830fe14a40443832d62720216dd5ce473930b2 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Fri, 25 Feb 2022 19:33:55 -0500
Subject: [PATCH 13/25] Prep nativebridge for emu-x86 sources

Use a different method for nativebridge cpu type detection

[nb] move to using PRODUCT_SYSTEM_DEFAULT_PROPERTIES

Revert "[nb] move to using PRODUCT_SYSTEM_DEFAULT_PROPERTIES"

This reverts commit ceab3d712f7df100a7ed8e750d247d047bf850db.

Use treble product properties for nativebridge

Add zygote64_32 to nativebridge script
---
 BoardConfig.mk                       |  5 ++++
 device.mk                            | 15 ++++++++++
 manifest.xml                         |  6 ++--
 nativebridge/Android.mk              | 10 +++++--
 nativebridge/bin/enable_nativebridge |  5 ++--
 nativebridge/nativebridge.mk         | 41 +++++++++++++++++++---------
 nativebridge/nativebridge64.mk       | 19 +++++++++++++
 7 files changed, 82 insertions(+), 19 deletions(-)
 create mode 100644 nativebridge/nativebridge64.mk

diff --git a/BoardConfig.mk b/BoardConfig.mk
index 196b05e..b5e44a5 100644
--- a/BoardConfig.mk
+++ b/BoardConfig.mk
@@ -112,3 +112,8 @@ BOARD_MESA3D_GALLIUM_DRIVERS := i915 crocus iris nouveau r300 r600 radeonsi svga
 BOARD_MESA3D_VULKAN_DRIVERS := amd intel
 #BOARD_MESA3D_BUILD_LIBGBM := true
 BOARD_USES_GRALLOC_HANDLE := true
+
+-include vendor/google/emu-x86/board/native_bridge_arm_on_x86.mk
+-include vendor/google/chromeos-x86/board/native_bridge_arm_on_x86.mk
+
+PRODUCT_FULL_TREBLE := true
diff --git a/device.mk b/device.mk
index 342bd45..5a4e806 100644
--- a/device.mk
+++ b/device.mk
@@ -131,7 +131,22 @@ GAPPS_VARIANT ?= pico
 $(call inherit-product-if-exists,$(if $(wildcard vendor/google/products/gms.mk),vendor/google/products/gms.mk,vendor/gapps/common/common-vendor.mk))
 
 # Get native bridge settings
+ifeq ($(patsubst %x86,,$(lastword $(TARGET_PRODUCT))),)
 $(call inherit-product-if-exists,$(LOCAL_PATH)/nativebridge/nativebridge.mk)
+endif
+
+ifeq ($(patsubst %x86_64,,$(lastword $(TARGET_PRODUCT))),)
+$(call inherit-product-if-exists,$(LOCAL_PATH)/nativebridge/nativebridge.mk)
+$(call inherit-product-if-exists,$(LOCAL_PATH)/nativebridge/nativebridge64.mk)
+endif
+
+$(call inherit-product-if-exists, vendor/google/emu-x86/target/libndk_translation.mk)
+$(call inherit-product-if-exists, vendor/google/emu-x86/target/native_bridge_arm_on_x86.mk)
+$(call inherit-product-if-exists, vendor/google/emu-x86/target/widevine.mk)
+
+$(call inherit-product-if-exists, vendor/google/chromeos-x86/target/houdini.mk)
+$(call inherit-product-if-exists, vendor/google/chromeos-x86/target/native_bridge_arm_on_x86.mk)
+$(call inherit-product-if-exists, vendor/google/chromeos-x86/target/widevine.mk)
 
 $(call inherit-product,$(if $(wildcard $(PRODUCT_DIR)packages.mk),$(PRODUCT_DIR),$(LOCAL_PATH)/)packages.mk)
 
diff --git a/manifest.xml b/manifest.xml
index d2fa470..c6cf5a9 100644
--- a/manifest.xml
+++ b/manifest.xml
@@ -159,8 +159,10 @@
             <name>IDrmFactory</name>
             <instance>default</instance>
         </interface>
-        <fqname>@1.2::ICryptoFactory/clearkey</fqname>
-        <fqname>@1.2::IDrmFactory/clearkey</fqname>
+        <fqname>@1.4::ICryptoFactory/clearkey</fqname>
+        <fqname>@1.4::IDrmFactory/clearkey</fqname>
+        <fqname>@1.4::ICryptoFactory/widevine</fqname>
+        <fqname>@1.4::IDrmFactory/widevine</fqname>
     </hal>
     <hal format="hidl">
         <name>android.hardware.gnss</name>
diff --git a/nativebridge/Android.mk b/nativebridge/Android.mk
index 90570b8..7cd7f16 100644
--- a/nativebridge/Android.mk
+++ b/nativebridge/Android.mk
@@ -19,9 +19,15 @@ LOCAL_CPPFLAGS := -std=c++11
 LOCAL_SHARED_LIBRARIES := libcutils libdl liblog
 LOCAL_HEADER_LIBRARIES := jni_headers libnativebridge-headers
 LOCAL_MULTILIB := both
-LOCAL_POST_INSTALL_CMD := $(hide) \
+ifneq ("$(wildcard vendor/google/chromeos-x86/proprietary/*)","")
+	include $(BUILD_SHARED_LIBRARY)
+else ifneq ("$(wildcard vendor/google/emu-x86/proprietary/*)","")
+	include $(BUILD_SHARED_LIBRARY)
+else
+	LOCAL_POST_INSTALL_CMD := $(hide) \
     rm -rf $(TARGET_OUT)/{bin,lib}/{arm*,*houdini*} {$(TARGET_OUT),$(PRODUCT_OUT)}/vendor/{bin,lib}/{arm*,*houdini*}; \
     mkdir -p $(TARGET_OUT)/{lib/arm,$(if $(filter true,$(TARGET_IS_64_BIT)),lib64/arm64)}; \
     touch $(TARGET_OUT)/lib/libhoudini.so $(if $(filter true,$(TARGET_IS_64_BIT)),$(TARGET_OUT)/lib64/libhoudini.so)
 
-include $(BUILD_SHARED_LIBRARY)
+	include $(BUILD_SHARED_LIBRARY)
+endif
diff --git a/nativebridge/bin/enable_nativebridge b/nativebridge/bin/enable_nativebridge
index 00da6f1..1cc329b 100644
--- a/nativebridge/bin/enable_nativebridge
+++ b/nativebridge/bin/enable_nativebridge
@@ -15,8 +15,7 @@ urls[5]=http://tinyurl.com/y6g6cx27
 
 mount_sfs()
 {
-	lodevice=`losetup -r -f -s $1`
-	[ -n "$lodevice" ] && mount -o 'ro,context="u:object_r:system_file:s0"' $lodevice $2
+	mount -o 'ro,context="u:object_r:system_file:s0"' $1 $2
 }
 
 if [ -z "$1" ]; then
@@ -83,4 +82,6 @@ else
 	log -pi -thoudini "houdini$1 enabled"
 fi
 
+[ "$(getprop ro.zygote)" = "zygote64_32" -a -z "$1" ] && exec $0 64
+
 exit 0
diff --git a/nativebridge/nativebridge.mk b/nativebridge/nativebridge.mk
index 69b14f0..8b0b600 100644
--- a/nativebridge/nativebridge.mk
+++ b/nativebridge/nativebridge.mk
@@ -11,8 +11,34 @@
 # Enable native bridge
 WITH_NATIVE_BRIDGE := true
 
+ifneq ("$(wildcard vendor/intel/houdini/*)","")
+HOUDINI_PREINSTALL := intel
+else ifneq ("$(wildcard vendor/google/chromeos-x86/proprietary/*)","")
+PRODUCT_PRODUCT_PROPERTIES := ro.dalvik.vm.native.bridge=libhoudini.so
+USE_CROS_HOUDINI_NB := true
+else ifneq ("$(wildcard vendor/google/emu-x86/proprietary/*)","")
+PRODUCT_PRODUCT_PROPERTIES := ro.dalvik.vm.native.bridge=libndk_translation.so
+USE_LIBNDK_TRANSLATION_NB := true
+else
+PRODUCT_PRODUCT_PROPERTIES := ro.dalvik.vm.native.bridge=libnb.so
+PRODUCT_PACKAGES := libnb
+endif
+
+# Native Bridge apps
+PRODUCT_PACKAGES += \
+    native_bridge_stub_library_defaults \
+    libnativebridge-headers \
+    libnativeloader-headers \
+    libqemupipe \
+    libandroidemu
+
+PRODUCT_HOST_PACKAGES, += \
+    libnativebridge \
+    libnativeloader
+
 # Native Bridge ABI List
 NATIVE_BRIDGE_ABI_LIST_32_BIT := armeabi-v7a armeabi
+NATIVE_BRIDGE_ABI_LIST := x86 armeabi-v7a armeabi
 
 LOCAL_SRC_FILES := bin/enable_nativebridge
 
@@ -23,18 +49,7 @@ PRODUCT_COPY_FILES := $(foreach f,$(LOCAL_SRC_FILES),$(LOCAL_PATH)/$(f):system/$
 
 PRODUCT_PROPERTY_OVERRIDES := \
     ro.dalvik.vm.isa.arm=x86 \
-    ro.enable.native.bridge.exec=1 \
-
-ifeq ($(TARGET_SUPPORTS_64_BIT_APPS),true)
-PRODUCT_PROPERTY_OVERRIDES += \
-    ro.dalvik.vm.isa.arm64=x86_64 \
-    ro.enable.native.bridge.exec64=1
-endif
-
-ifneq ($(HOUDINI_PREINSTALL),intel)
-PRODUCT_PRODUCT_PROPERTIES := ro.dalvik.vm.native.bridge=libnb.so
-
-PRODUCT_PACKAGES := libnb
-endif
+    ro.enable.native.bridge.exec=1 
 
 $(call inherit-product-if-exists,vendor/intel/houdini/houdini.mk)
+
diff --git a/nativebridge/nativebridge64.mk b/nativebridge/nativebridge64.mk
new file mode 100644
index 0000000..3f16413
--- /dev/null
+++ b/nativebridge/nativebridge64.mk
@@ -0,0 +1,19 @@
+#
+# Copyright (C) 2021 The Android-x86 Open Source Project
+#
+# Licensed under the GNU General Public License Version 2 or later.
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.gnu.org/licenses/gpl.html
+#
+
+NATIVE_BRIDGE_ABI_LIST_64_BIT := arm64-v8a
+NATIVE_BRIDGE_ABI_LIST := x86_64 x86 arm64-v8a armeabi-v7a armeabi
+TARGET_CPU_ABI_LIST := x86_64 x86 arm64-v8a armeabi-v7a armeabi
+
+PRODUCT_PROPERTY_OVERRIDES += \
+    ro.dalvik.vm.isa.arm64=x86_64 \
+    ro.enable.native.bridge.exec64=1
+
+
-- 
2.25.1

